### PR TITLE
Support Python3.11

### LINKED ISSUE #1245: 3.11 deprecation: reuse_address in loop.create_datagram_endpoint
>     self.protocol, self.endpoint = await self.loop.create_datagram_endpoint(
>                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> TypeError: BaseEventLoop.create_datagram_endpoint() got an unexpected keyword argument 'reuse_address'

asyncio [docs](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.create_datagram_endpoint):

> Changed in version 3.11: The `reuse_address` parameter, disabled since Python 3.9.0, 3.8.1, 3.7.6 and 3.6.10, has been entirely removed.

PR incoming

### PR DESCRIPTION
**Pull request by alexrudd2**
Closes #1245.

All tests pass on Python3.11 the same as Python3.10 and 3.8 on MacOS for me.

I see you have simplified and cleaned up the CI a lot; awesome! Hopefully it's easy to add 3.11

### CODE REVIEW

File: examples/contrib/serial_forwarder.py [hunk 1]
```diff
@@ -42,9 +42,7 @@ async def run(self):
         for i in slaves:
             store[i] = RemoteSlaveContext(client, unit=i)
         context = ModbusServerContext(slaves=store, single=False)
-        self.server = ModbusTcpServer(
-            context, address=(server_ip, server_port), allow_reuse_address=True
-        )
+        self.server = ModbusTcpServer(context, address=(server_ip, server_port))
```
Comment (janiversen) [inline examples/contrib/serial_forwarder.py:45]: Please do not remove it from TCP it is needed. It is only removed from UDP.

File: examples/server_async.py [hunk 2]
```diff
@@ -151,7 +151,6 @@ async def run_async_server(args):
             # custom_functions=[],  # allow custom handling
             framer=args.framer,  # The framer strategy to use
             # handler=None,  # handler for each session
-            allow_reuse_address=True,  # allow the reuse of an address
```
Comment (janiversen) [inline examples/server_async.py:154]: It is needed for TCP.

File: examples/server_async.py [hunk 3]
```diff
@@ -167,7 +166,6 @@ async def run_async_server(args):
             # custom_functions=[],  # allow custom handling
             framer=args.framer,  # The framer strategy to use
             # handler=None,  # handler for each session
-            # TBD allow_reuse_address=True,  # allow the reuse of an address
```
Comment (janiversen) [inline examples/server_async.py:170]: Needed for TCP.

And all similar places, not UDP (I will not comment on the rest).

File: examples/server_async.py [hunk 4]
```diff
@@ -213,7 +211,6 @@ async def run_async_server(args):
             address=address,  # listen address
             framer=args.framer,  # The framer strategy to use
             # handler=None,  # handler for each session
-            allow_reuse_address=True,  # allow the reuse of an address
```
Comment (janiversen) [inline examples/server_async.py:216]: Needed for TCP.

**Review by janiversen — CHANGES_REQUESTED** [review]
Removing reuse_addr for UDP is a good thing, I thought I had done that already. BUT you cannot remove reuse_addr from TCP, this is highly needed.

**Comment by janiversen** [comment]
IMPORTANT, please do a "git pull" before making changes, I added a commit changing README.rst

**Comment by alexrudd2** [comment]
> Removing reuse_addr for UDP is a good thing, I thought I had done that already. 

You had set it to `False`, which now emits a DeprecationWarning in 3.11.

> BUT you cannot remove reuse_addr from TCP, this is highly needed.

OK, I assume TLS follows TCP.  Changed to only UDP.

> IMPORTANT, please do a "git pull" before making changes, I added a commit changing README.rst

OK.  I rebased and force-pushed.

**Review by janiversen — APPROVED** [review]
LGTM.

I wrote earlier "IMPORTANT, please do a "git pull" before making changes, I added a commit changing README.rst"

You did not do that, but instead did "git push -f" and thus killed my commit. It would have been nice, it you had included the change in your commit, so I avoided doing it twice.

**Comment by alexrudd2** [comment]
My apologies for not understanding your message, although I didn't ignore it.

 I saw you had committed 33659da, which merged `dev` into my branch.  (It's still available with `git log`)  I did not see any relevant README changes that weren't in other commits - the changes present would get dropped by merging this back into `dev`.  

I pulled `dev` and rebased this branch on top so there were no conflicts, then did `git push --force-with-lease`.  Using this instead of `--force` means I had was not overwriting any remote commits I had not seen. 

Had you encountered a merge conflict?  Did I make a mistake in the rebase?

**Comment by janiversen** [comment]
I did a commit on this branch not dev. I did it directly on dev in the second turn.

I did not encounter a merge conflict my commit was simply gone, due to your “git push —force…” (any version of —force, replaces the commit log for the branch on the server with your version).

You should have done “git pull” before starting to work, then the normal “git push” would have worked. But anyhow it is history now and it was not a big problem, just wanted to mention it.

Looking forward to see more work from you. Let me know if you want ideas of what needs to be done.

**Pull request merged**